### DISCUSSION TITLE
Need a variety of font sizes

### DISCUSSION THREAD

**Discussion by griffinmt** [Bodmer/TFT_eSPI#2291] [Q&A - Fonts]
Using the current version of TFT_eSPI (2.4.79) with my ILI9488 and ESP32 on Arduino ide V1.8.18
My problem is that using the font sizes set at 1 or 2 or 3 doesnt give me what I need in all cases. Logically I need sizes that would be the equivilent of 1.8 and 2.5. How do I acheive this without major code changes to the library code.
TIA
Martyn

**Comment by Bodmer** [comment]
It is not clear which font you mean. Only integer size multipliers are possible with the built-in fonts.  The library supports quite a few font formats that provide different font styles and sizes. Users can also create fonts for the "Free font" and "Smooth font" formats supported. There are examples included with the library for these.

This library extension provides access to more than 3000 font files of various styles and sizes, so that may be suitable.

The "ultimate" font type for resizng is the Truetype, this requires the [OpenFontRender ](https://github.com/takkaO/OpenFontRender )library. However the print placement options are limited at the moment.

**Comment by griffinmt** [comment]
Thanks for your answer. I guess i need to do a lot more digging 😏.

**Reply by Bodmer** [reply]
OK, I see I misses a hyperlink above. Should have been:

This library extension provides access to more than 3000 font files of various styles and sizes, so that may be suitable:
https://github.com/Bodmer/TFT_eSPI_ext

**Reply by Bodmer** [reply]
If you need specific help the post back.
